components/list/list-item.js

```diff
@@ -228,8 +228,20 @@ class ListItem extends RtlMixin(LitElement) {
 			</div>
 		`;
 	}
-
 	updated(changedProperties) {
+		if (changedProperties.has('key')) {
+			const oldValue = changedProperties.get('key');
+			if (typeof oldValue !== 'undefined') {
+				this.setIsSelected(undefined, true);
+				this._fireItemSelected(false, oldValue);
+			}
+		}
+
+		if (changedProperties.has('selected')) {
+			const checkBox = this.shadowRoot.querySelector(`#${this._checkBoxId}`);
+			checkBox && (checkBox.checked = this.selected);
```

Comment:
I was able to remove the getter and setters for selected. However, I was not able to remove this bit. Using `?checked=${this.selected}` causes issue where the checkbox will remained checked even if `checked` is not an attribute of the checkbox.

Assuming `:checked` css property isn't being updated.